tools/download-bootstrap.R

```diff
@@ -0,0 +1,73 @@
+owd <- setwd('inst/rmd/h/bootstrap/css/')
+unlink('*.min.css')
+
+local({
+  themes <- jsonlite::fromJSON('https://bootswatch.com/api/4.json')$themes
+  for (i in seq_len(nrow(themes))) xfun::download_file(
+    themes[i, 'cssMin'], paste0(tolower(themes[i, 'name']), '.min.css')
+  )
+})
+
+cdn <- function(...) paste(
+  c('https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/', ...), collapse = ''
+)
+
+xfun::download_file(cdn('css/bootstrap.min.css'))
+# bootstrap.js + popper.js
+xfun::download_file(cdn('js/bootstrap.bundle.min.js'), '../js/bootstrap.min.js')
+
+# replace the CSS @import with actual CSS font specs so the themes can work offline
+local({
+  # font weights
+  w <- c(
+    '100' = 'Thin', '200' = 'ExtraLight', '300' = 'Light', '400' = 'Regular',
+    '500' = 'Medium', '600' = 'SemiBold', '700' = 'Bold', '800' = 'ExtraBold',
+    '900' = 'Black'
+  )
+  w <- c(w, setNames(paste(w, 'Italic'), paste0(names(w), 'italic')))
+  w['400italic'] <- 'Italic'
+  fonts <- list.files('fonts')
```

Comment:
Where is `fonts/` coming from? [bootscss](https://github.com/rstudio/bootscss) uses this [CLI tool](https://github.com/neverpanic/google-font-download) (like shinythemes [does](https://github.com/rstudio/shinythemes/blob/master/inst/shinythemes/fonts/README.md)) which leads to about 1.3MB in ttf files whereas this directory is 3.8MB. Also note that the CLI tool generates the CSS font spec for you...

Reply:
Good to know! Thanks for the tip! The fonts folder exists in the current version of rmarkdown. I guess JJ created it initially. I downloaded a few new fonts for this PR.